platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/AssertionConsumerService.java

```diff
@@ -200,11 +200,16 @@ public Response getSamlResponse(
       @QueryParam(SIG_ALG) String signatureAlgorithm,
       @QueryParam(SIGNATURE) String signature) {
 
-    LOGGER.info("HTTP-Redirect binding should not be used for Single Sign On responses");
+    LOGGER.warn("HTTP-Redirect binding should not be used for Single Sign-On responses");
     if (validateSignature(deflatedSamlResponse, relayState, signatureAlgorithm, signature)) {
       try {
-        return processSamlResponse(
-            RestSecurity.inflateBase64(deflatedSamlResponse), relayState, signature != null);
+        return (StringUtils.isNotBlank(deflatedSamlResponse)
+                && StringUtils.isNotBlank(relayState)
+                && StringUtils.isNotBlank(signatureAlgorithm))
```

Comment:
✏️ You only need to check for `deflatedSamlResponse` being blank here. RelayState being null is accounted for further down the chain, and signature algorithm isn't used anymore (it's used above).

Reply:
I'm confused about the logic of validateSignature. Why does it always return true if the signature is not null?

Reply:
It doesn't. ValidateSignature would return false if signature is not null, but one of the other parameters is blank. It will also return false if this call returns false: https://github.com/codice/ddf/blob/b1d6bb59251c6bbf2b9919bff630d4820abe04f7/platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/AssertionConsumerService.java#L240-L245